src/smk/layer/layer-vector.js

```diff
@@ -21,54 +21,82 @@ include.module( 'layer.layer-vector-js', [ 'layer.layer-js' ], function () {
         var cv = $( '<canvas width="' + width * mult + '" height="' + height + '">' ).get( 0 )
         var ctx = cv.getContext( '2d' )
 
-        var styles = [].concat( self.config.style )
+        const legendData = [];
+        legendData.push({
+            title: self.config.legend.title || self.config.title,
+            style: self.config.style
+        });
+
+        if (self.config.conditionalStyles) {
+            self.config.conditionalStyles.forEach(cs => {
```

Comment:
I haven't tried running this code, but it seems odd to have `self.config.conditionalStyles.foreach()` nested within another  `self.config.conditionalStyles.foreach()` ie line 33 nested within 31.

Reply:
Probably a copy/paste error ... this change went through a lot of iterations before I pushed it to review. I'll update so there's only one.